examples/seismic/viscoacoustic/operators.py

```diff
@@ -543,6 +546,101 @@ def AdjointOperator(model, geometry, space_order=4, kernel='sls', time_order=2,
     return Operator(eqn + srcrec, subs=model.spacing_map, name='Adjoint', **kwargs)
 
 
+def GradientOperator(model, geometry, space_order=4, kernel='sls', time_order=2,
+                     save=True, **kwargs):
+    """
+    Construct a gradient operator in an acoustic media.
+
+    Parameters
+    ----------
+    model : Model
+        Object containing the physical parameters.
+    geometry : AcquisitionGeometry
+        Geometry object that contains the source (SparseTimeFunction) and
+        receivers (SparseTimeFunction) and their position.
+    space_order : int, optional
+        Space discretization order.
+    save : int or Buffer, optional
+        Option to store the entire (unrolled) wavefield.
+    kernel : selects a visco-acoustic equation from the options below:
+        sls (Standard Linear Solid) :
+        1st order - Blanch and Symes (1995) / Dutta and Schuster (2014)
+        viscoacoustic equation
+        2nd order - Bai et al. (2014) viscoacoustic equation
+        ren - Ren et al. (2014) viscoacoustic equation
+        deng_mcmechan - Deng and McMechan (2007) viscoacoustic equation
+        Defaults to sls 2nd order.
+    """
+    # Gradient symbol and wavefield symbols
+    grad = Function(name='grad', grid=model.grid)
+    p = TimeFunction(name='p', grid=model.grid, save=geometry.nt if save
+                     else None, time_order=2, space_order=space_order,
+                     staggered=NODE)
+    pa = TimeFunction(name='pa', grid=model.grid, save=None,
+                      time_order=time_order, space_order=space_order, staggered=NODE)
+
+    # Equations kernels
+    eq_kernel = kernels[kernel]
+    eqn = eq_kernel(model, geometry, pa, forward=False, save=save, **kwargs)
+
+    gradient_update = Inc(grad, - p.dt2 * pa)
+
+    # Add expression for receiver injection
+    _, recterm = src_rec(pa, model, geometry, forward=False)
+
+    # Substitute spacing terms to reduce flops
+    return Operator(eqn + recterm + [gradient_update], subs=model.spacing_map,
+                    name='Gradient', **kwargs)
+
+
+def BornOperator(model, geometry, space_order=4, kernel='sls', time_order=2, **kwargs):
+    """
+    Construct an Linearized Born operator in an acoustic media.
+
+    Parameters
+    ----------
+    model : Model
+        Object containing the physical parameters.
+    geometry : AcquisitionGeometry
+        Geometry object that contains the source (SparseTimeFunction) and
+        receivers (SparseTimeFunction) and their position.
+    space_order : int, optional
+        Space discretization order.
+    kernel : str, optional
+        Type of discretization, centered or shifted.
+    """
+    # Create wavefields and a dm field
+    p = TimeFunction(name='p', grid=model.grid, save=None,
+                     time_order=time_order, space_order=space_order, staggered=NODE)
+    P = TimeFunction(name='P', grid=model.grid, save=None,
```

Comment:
Maybe a slightly different naming could be more clear here? The `p` vs `P` is probably not ideal imo

Reply:
Thanks for the review.
We did it that way because we were based on the acoustic example. There ( in `operator.py`), `u` and `U` are used to represent the background and scattered wavefields, respectively.
The difference is that in our case we are using `p` and `P`.